ide/coqide/idetop.ml

```diff
@@ -202,38 +208,42 @@ let process_goal_diffs diff_goal_map oldp nsigma ng =
     Interface.goal_id = Proof.goal_uid ng; Interface.goal_name = name }
 
 let export_pre_goals Proof.{ sigma; goals; stack } process =
+  let filter_gls evds = List.filter (fun e -> Evd.mem sigma e) evds in
   let process = List.map (process sigma) in
   { Interface.fg_goals       = process goals
-  ; Interface.bg_goals       = List.(map (fun (lg,rg) -> process lg, process rg)) stack
+  ; Interface.bg_goals       = List.(map (fun (lg,rg) -> process (filter_gls lg), process (filter_gls rg))) stack
```

Comment:
I added this filtering to try to handle this script, but it produces a different message when at a debugger breakpoint because it's using an out of date `stack` (generated by the highlighted code in `tactic_debug`).

![image](https://user-images.githubusercontent.com/1253341/146843619-e70ad605-b107-4082-b213-1fa14731d75f.png)

Reply:
PS It works for other cases that don't need the latest `stack`:

![image](https://user-images.githubusercontent.com/1253341/146844879-0b2e19ab-2db7-42c2-a539-7405cb3f0504.png)